File: go/exercise39/sorting_records.go
package main

import (
	"database/sql"
	"fmt"

	_ "github.com/mattn/go-sqlite3"
)

const file string = "./employees.db"

func main() {
	type Employee struct {
		FirstName string
		LastName  string
		Position  string
		Date      string
	}
	employees := []Employee{}

	db, err := sql.Open("sqlite3", file)
	if err != nil {
		fmt.Println(err)
		return
	}
	defer db.Close()

	rows, err := db.Query("SELECT * from employees")
	if err != nil {
		fmt.Println(err)
		return
	}

	for rows.Next() {
		var emp Employee

		rows.Scan(&emp.FirstName, &emp.LastName, &emp.Position, &emp.Date)
		employees = append(employees, emp)
	}

	fmt.Printf("%-19s | %-17s | %s\n", "Name", "Position", "Separation Date")
	fmt.Println("-----------------------------------------------------")
	for _, emp := range employees {
		fmt.Printf("%-19s | %-17s | %s\n",
			emp.FirstName+" "+emp.LastName,
			emp.Position,
			emp.Date,
		)
	}
}

File: javascript/exercise39/sorting-records.js
import Database from "better-sqlite3";

const db = new Database("records.db", { fileMustExist: true });
db.pragma("journal_mode = WAL");

/** @type [{first_name: string, last_name: string, position: string, date: string}] rows */
const rows = db.prepare("SELECT * FROM employees").all();
db.close(); // close db

rows.sort((a, b) => a.last_name.localeCompare(b.last_name));

console.log(
	`${"Name".padEnd(20)} | ${"Position".padEnd(20)} | Separation Date`,
);
console.log("-".repeat(60));
for (const row of rows) {
	const fullname = `${row.first_name} ${row.last_name}`;
	console.log(
		`${fullname.padEnd(20)} | ${row.position.padEnd(20)} | ${row.date || ""}`,
	);
}
